minwin.go

```diff
@@ -0,0 +1,572 @@
+		}
+
+		if !mw.createPublisher.Publish() {
+			return ^uintptr(0)
+		}
+		return 0
+	case win.WM_ACTIVATE:
+		if wParam > 0 {
+			mw.activatePublisher.Publish()
+		} else {
+			mw.deactivatePublisher.Publish()
+		}
+		return 0
+	case win.WM_ERASEBKGND:
+		return 0
+	case win.WM_DPICHANGED:
+		dpi := int(win.LOWORD(uint32(wParam)))
+		mw.dpiChangedPublisher.Publish(dpi)
+		newRect := (*win.RECT)(unsafe.Pointer(lParam))
+		win.SetWindowPos(
+			hwnd,
+			0,
+			newRect.Left,
+			newRect.Top,
+			newRect.Width(),
+			newRect.Height(),
+			win.SWP_NOACTIVATE|win.SWP_NOZORDER,
+		)
+		return 0
+	case win.WM_DESTROY:
+		mw.destroyPublisher.Publish()
+		return 0
+	case win.WM_NCDESTROY:
+		mw.hWnd = 0
+		delete(minWins, mw)
+	case win.WM_WINDOWPOSCHANGED:
+		wp := (*win.WINDOWPOS)(unsafe.Pointer(lParam))
+		if (wp.Flags & win.SWP_SHOWWINDOW) != 0 {
+			mw.visibilityChangedPublisher.Publish(true)
+		} else if (wp.Flags & win.SWP_HIDEWINDOW) != 0 {
+			mw.visibilityChangedPublisher.Publish(false)
+		}
+
+		if (wp.Flags & win.SWP_NOMOVE) == 0 {
+			mw.movePublisher.Publish(Point{X: int(wp.X), Y: int(wp.X)})
+		}
+
+		if (wp.Flags & win.SWP_NOSIZE) == 0 {
+			sizePx := Size{Width: int(wp.Cx), Height: int(wp.Cy)}
+			mw.sizePublisher.Publish(SizeTo96DPI(sizePx, mw.DPI()))
+		}
+		return 0
+	case win.WM_SETTEXT:
+		result := mw.defWindowProc(hwnd, msg, wParam, lParam)
+		if result != 0 {
+			newText := windows.UTF16PtrToString((*uint16)(unsafe.Pointer(lParam)))
+			mw.textChangedPublisher.Publish(newText)
+		}
+		return result
+	}
+
+	return mw.defWindowProc(hwnd, msg, wParam, lParam)
+}
+
+func registerMinWinClass() (className16 *uint16, err error) {
+	className16, err = windows.UTF16PtrFromString(minWinClassName)
+	if err != nil {
+		return nil, err
+	}
+
+	if registeredWindowClasses[minWinClassName] != nil {
+		return className16, nil
+	}
+
+	if minWinProcCb == 0 {
+		minWinProcCb = windows.NewCallback(minWinProc)
+	}
+
+	wc := win.WNDCLASSEX{
+		CbSize:        uint32(unsafe.Sizeof(win.WNDCLASSEX{})),
+		Style:         win.CS_DBLCLKS, // We don't bother with redrawing since we're just a host window.
+		LpfnWndProc:   minWinProcCb,
+		HCursor:       win.LoadCursor(0, win.MAKEINTRESOURCE(win.IDC_ARROW)),
+		LpszClassName: className16,
+	}
+	if win.RegisterClassEx(&wc) == 0 {
+		return nil, lastError("RegisterClassEx")
+	}
+
+	registeredWindowClasses[minWinClassName] = newWndClassInfo()
+	return className16, err
+}
+
+// SetFocus sets keyboard focus to focusTo, a window contained within topLevel.
+// It tries to respect walk focus-setting conventions when possible, otherwise
+// it falls back to calling [win.SetFocus] directly.
+func (mw *MinWin) SetFocus(topLevel, focusTo win.HWND) {
+	if topLevelWindow := windowFromHandle(topLevel); topLevelWindow != nil {
+		if form, ok := topLevelWindow.(Form); ok {
+			if focusToWindow := windowFromHandle(focusTo); focusToWindow != nil {
+				form.SetFocusToWindow(focusToWindow)
+				return
+			}
+		}
+	}
```

Comment:
Maybe this? A checked type assertion from a nil interface is fine.
```suggestion
	if form, ok := windowFromHandle(topLevel).(Form); ok {
		if focusToWindow := windowFromHandle(focusTo); focusToWindow != nil {
			form.SetFocusToWindow(focusToWindow)
			return
		}
	}
```